src/Component/ContactDetails.elm

```diff
@@ -0,0 +1,537 @@
+toContactMethod contactMethod =
+    case contactMethod of
+        "Email" ->
+            Just Email
+
+        "Post" ->
+            Just Post
+
+        "Phone" ->
+            Just Phone
+
+        _ ->
+            Nothing
+
+
+fromContactMethod : PreferredContactMethod -> String
+fromContactMethod contactMethod =
+    case contactMethod of
+        Email ->
+            "E-mail"
+
+        Post ->
+            "Post"
+
+        Phone ->
+            "Phone"
+
+
+preferredContactMethodSelect : Model -> List (Html Msg)
+preferredContactMethodSelect model =
+    if model.isInherited then
+        [ span [] [ text <| fromContactMethod model.selectedContactMethod ] ]
+
+    else
+        [ select
+            [ name "contact"
+            , id "contact"
+            , autocomplete True
+            , onInput <| toContactMethod >> Maybe.withDefault model.selectedContactMethod >> SelectedContactMethodChanged
+            , class "col-start-1 row-start-1 w-full appearance-none rounded-md bg-white py-1.5 pl-3 pr-8 text-base text-gray-900 outline outline-1 -outline-offset-1 outline-gray-300 focus:outline focus:outline-2 focus:-outline-offset-2 focus:outline-indigo-600 sm:text-sm/6"
+            ]
+            [ option [ value "Email" ] [ text "E-mail" ]
+            , option [ value "Post" ] [ text "Post" ]
+            , option [ value "Phone" ] [ text "Phone" ]
+            ]
+        , div [ class "pointer-events-none absolute inset-y-0 right-0 flex items-center px-2 text-gray-700" ]
+            [ Svg.svg
+                [ SvgAttrs.class "fill-current h-4 w-4"
+                , SvgAttrs.viewBox "0 0 20 20"
+                ]
+                [ Svg.path [ SvgAttrs.d "M9.293 12.95l.707.707L15.657 8l-1.414-1.414L10 10.828 5.757 6.586 4.343 8z" ] [] ]
+            ]
+        ]
+
+
+introductionText : String
+introductionText =
+    """This form contains contact details for the user group. Select a preferred way of contact and make sure that
+    the corresponding information is filled in. """
+
+
+inheritedText : Bool -> List (Html Msg)
+inheritedText isInherited =
+    if isInherited then
+        [ text "The current settings are "
+        , span [ class "font-bold" ] [ text "inherited" ]
+        , text " from a parent user group, meaning that they can not be modified."
+        ]
+
+    else
+        []
+
+
+view : Model -> Html Msg
+view model =
+    let
+        isInherited =
+            model.isInherited
+    in
+    div []
+        [ p [ class "mb-8 text-sm" ] <| text introductionText :: inheritedText model.isInherited
+        , div [ class "mt-10 grid grid-cols-1 gap-x-6 sm:grid-cols-6" ]
+            [ div [ class "sm:col-span-3 h-0 mb-24" ]
+                [ label [ for "contact", class "block text-gray-700 text-sm font-bold mb-2" ] [ text "Preferred contact method" ]
+                , div [ class "mt-2 relative" ] <| preferredContactMethodSelect model
+                ]
+            , div [ class "col-span-full h-0 mb-24" ] <|
```

Comment:
Is there an advantage in having this `div [ class "col-span-full h-0 mb-24" ]` repeated per each input element rather than, say, put directly inside `contactDetailTextForm`